airbyte-metrics/reporter/src/main/java/io/airbyte/metrics/reporter/MetricRepository.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.reporter;
+
+import static io.airbyte.db.instance.configs.jooq.generated.Tables.CONNECTION;
+import static io.airbyte.db.instance.jobs.jooq.generated.Tables.JOBS;
+import static org.jooq.impl.SQLDataType.VARCHAR;
+
+import io.airbyte.db.instance.configs.jooq.generated.enums.StatusType;
+import io.airbyte.db.instance.jobs.jooq.generated.enums.JobStatus;
+import jakarta.inject.Singleton;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.jooq.DSLContext;
+
+@Singleton
+class MetricRepository {
+
+  private final DSLContext ctx;
+
+  MetricRepository(final DSLContext ctx) {
+    this.ctx = ctx;
+  }
+
+  int numberOfPendingJobs() {
+    return ctx.selectCount()
+        .from(JOBS)
+        .where(JOBS.STATUS.eq(JobStatus.pending))
+        .fetchOne(0, int.class);
+  }
+
+  int numberOfRunningJobs() {
+    return ctx.selectCount()
+        .from(JOBS)
+        .join(CONNECTION)
+        .on(CONNECTION.ID.cast(VARCHAR(255)).eq(JOBS.SCOPE))
+        .where(JOBS.STATUS.eq(JobStatus.running).and(CONNECTION.STATUS.eq(StatusType.active)))
+        .fetchOne(0, int.class);
+  }
+
+  int numberOfOrphanRunningJobs() {
+    return ctx.selectCount()
+        .from(JOBS)
+        .join(CONNECTION)
+        .on(CONNECTION.ID.cast(VARCHAR(255)).eq(JOBS.SCOPE))
+        .where(JOBS.STATUS.eq(JobStatus.running).and(CONNECTION.STATUS.ne(StatusType.active)))
+        .fetchOne(0, int.class);
+  }
+
+  long oldestPendingJobAgeSecs() {
+    return oldestJobAgeSecs(JobStatus.pending);
+  }
+
+  long oldestRunningJobAgeSecs() {
+    return oldestJobAgeSecs(JobStatus.running);
+  }
+
+  List<Long> numberOfActiveConnPerWorkspace() {
+    final var query = """
+                      SELECT workspace_id, count(c.id) as num_conn
+                      FROM actor
+                        INNER JOIN workspace ws ON actor.workspace_id = ws.id
+                        INNER JOIN connection c ON actor.id = c.source_id
+                      WHERE ws.tombstone = false
+                        AND actor.tombstone = false AND actor.actor_type = 'source'
+                        AND c.status = 'active'
+                      GROUP BY workspace_id;
+                      """;
+    return ctx.fetch(query).getValues("num_conn", long.class);
+  }
+
+  long numScheduledActiveConnectionsInLastDay() {
```

Comment:
Aren't we going to miss some connection here? expecially the cron and the manual ones that have been triggered.

Reply:
Possibly?  This was also copied verbatim from the [existing implementation](https://github.com/airbytehq/airbyte/pull/17365/files/477451ec7a0a567ca5351edc23caae8b59863e40#diff-5491910fea78d525278084231b60d98ea3242018c44d470e32fefa5df6ffe9f4L215-L227).  See my comments above about a follow-up PR.